CFUValidator/Validators/ZipContentsValidator.cs

```diff
@@ -57,9 +58,16 @@ public ZipContentsValidator(List<XenServerPatchAlert> alerts)
 
         protected override void ValidateCore(Action<string> statusReporter)
         {
-            foreach (XenServerPatchAlert alert in alerts.OrderBy(a => a.Patch.Name))
+            try
             {
-                DownloadPatchFile(alert, statusReporter);
+                TokenManager.GetToken(_configProvider);
+
+                foreach (XenServerPatchAlert alert in alerts.OrderBy(a => a.Patch.Name))

```

Comment:
Possibly, but I wouldn't address this right now because it's not code that has been changed by this PR, it was just rearranged to accommodate usage with a token.